dowhy/datasets.py

```diff
@@ -92,6 +92,150 @@ def linear_dataset(
+
+This function generates a pandas dataFrame with num_samples records. The variables follow a naming convention were the first letter indicates its role in the causality graph and then a sequence number.
+
+Parameters
+---------------
+Beta : int or list/ndarray of lenght num_treatments of type int
+	defines the causality effect of interest. i.e. the magnitude of the resulting ATE. It affects the value magnitude for the continuous treatments ('v?') and outcomes variables - 'y' . Note, the absolute value is used mostly to factor the ranges.
+num_common_causes : int
+num_samples : int
+num_instruments : int, default 0
+num_effect_modifiers : int, default 0
+num_treatments : int, default 1
+num_frontdoor_variables : int, default  0
+treatment_is_binary : bool, default True
+	cannot be True if treatment_is_category is True
+treatment_is_category : bool, default False
+	cannot be True if treatment_is_binary is True
+outcome_is_binary : bool, default False,
+stochastic_discretization : bool, default True
+num_discrete_common_causes : int, default 0
+          of the total num_common_causes
+	quartiles are used when discretised variables are specified. They can be hot encoded.
+num_discrete_instruments  : int , default 0
+	of the total num_instruments
+	quartiles are used when discretised variables are specified. They can be hot encoded.
+num_discrete_effect_modifiers : int, default 0
+	of the total effect_modifiers
+	quartiles are used when discretised variables are specified. They can be hot encoded.
+stddev_treatment_noise  : int, default 1
+stddev_outcome_noise : int , default 0.01
+one_hot_encode : bool, default False
+
+Returns
+----------
+Dict
+	"df": pd.dataFrame
+		    with num_samples records. The variables follow a naming convention were the first letter indicates its role in the causality graph and then a sequence number.
+	
+		v variables - are the treatments. They can be binary or continues. In the case of continues abs(*beta*) defines thier magnitude;
+		
+		y - is the outcome variable where abs(*beta*) defines its magnitude. Basically:
+		 y = normal(0, stddev_outcome_noise) + t @ beta [where @ is a numpy matrix multiplication allowing for beta be a vector]
+		
+		W variables - commonly cause both the treatment and the outcome and are iid. if continuous, they are Norm(mue = Unif(-1,1), sigma = 1)
+		
+		Z variables - Instrument variables. Each one affects all treatments. i.e. if there is one instrument and two treatments then z0->v0, z0->v1
+		
+		X variables - effect modifiers. If continues, they are Norm(mue = Unif(-1,1), sigma = 1)
+	"treatment_name": str/list(str)
+	"outcome_name": str
+	"common_causes_names": str/list(str)
+	"instrument_names": str/list(str)
+	"effect_modifier_names": str/list(str)
+	"frontdoor_variables_names": str/list(str)
+	"dot_graph": dot_graph,
+	"gml_graph": gml_graph,
+	"ate": float
+
+Examples
+------------
+
+>>> import networkx as nx
```

Comment:
for indenting code, sphinx uses the reST format and we'll need to use the same one. 
I'd suggest using the code-block syntax, as shown in https://stackoverflow.com/questions/64451966/python-sphinx-how-to-embed-code-into-a-docstring